src/main/java/org/springframework/hateoas/mvc/ControllerLinkBuilder.java

```diff
@@ -368,14 +380,72 @@ private static HttpServletRequest getCurrentRequest() {
 	@RequiredArgsConstructor
 	private static class CachingAnnotationMappingDiscoverer implements MappingDiscoverer {
```

Comment:
I find the name kind of misleading since the class seems to do two things:

1. determine the correct delegate to use.

2. caching `UriTemplates`

I see various ways to improve that: 2 classes, lengthy name or some JavaDoc.